code/modules/mob/living/carbon/human/species_types/mothmen.dm

```diff
@@ -47,3 +55,125 @@
+			to_chat(H, "<span class='warning'>You cannot weave a cocoon in your current state.</span>")
+			return
+		H.visible_message("<span class='notice'>[H] finishes weaving a cocoon!</span>", "<span class='notice'>You finish weaving your cocoon.</span>")
+		var/obj/structure/moth/cocoon/C = new(get_turf(H))
+		H.forceMove(C)
+		H.Sleeping(20, 0)
+		C.preparing_to_emerge = TRUE
+		H.apply_status_effect(STATUS_EFFECT_COCOONED)
+		//owner.log_message("[key_name(owner)] has finished weaving a cocoon at [AREACOORD(owner)]")
+		addtimer(CALLBACK(src, .proc/emerge, C), COCOON_EMERGE_DELAY, TIMER_UNIQUE)
+	else
+		to_chat(H, "<span class='warning'>You need to hold still in order to weave a cocoon!</span>")
+
+//Removes moth from cocoon, restores burnt wings
+/datum/action/innate/cocoon/proc/emerge(obj/structure/moth/cocoon/C)
+	for(var/mob/living/carbon/human/H in C.contents)
+		if(!H.has_status_effect(STATUS_EFFECT_COCOONED))
+			return
+		//if(H.dna.features["moth_wings"] == "Burnt Off") //this check seems redundant as burned wings are a roundstart option and sending a signal to clear a non-existing mood event doesn't cause any issues
+		SEND_SIGNAL(H, COMSIG_CLEAR_MOOD_EVENT, "burnt_wings")
+		//if(!ismoth(H))//mutation toxin is a thing, so this is here to prevent accidentally creating a mishmash of moth and whatever else
+			//return Note - this makes the transformed user NEVER wake up
+		if(ismoth(H))
+			REMOVE_TRAIT(H, TRAIT_MOTH_BURNT, "fire")
+		H.dna.species.handle_mutant_bodyparts(H)
+		H.dna.species.handle_body(H)
+	C.preparing_to_emerge = FALSE
+	qdel(C)
+
+/datum/status_effect/cocooned/tick()
+	owner.SetSleeping(20, TRUE)
+	owner.adjustBruteLoss(-(COCOON_HEAL_AMOUNT / (COCOON_EMERGE_DELAY)), 0)
+	owner.adjustFireLoss(-(COCOON_HEAL_AMOUNT / (COCOON_EMERGE_DELAY)), 0)
+	owner.adjust_nutrition(-(COCOON_NUTRITION_AMOUNT / (COCOON_EMERGE_DELAY)), 0)
+
+
+/obj/structure/moth/cocoon
+	name = "\improper Mothperson cocoon"
+	desc = "Someone wrapped in a Mothperson cocoon. It's best to let them rest."
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "cocoon_large1"
+	anchored = TRUE
+	max_integrity = 10
+	var/preparing_to_emerge
+
+/obj/structure/moth/cocoon/play_attack_sound(damage_amount, damage_type = BRUTE, damage_flag = 0)
+	switch(damage_type)
+		if(BRUTE)
+			playsound(src, 'sound/weapons/slash.ogg', 80, TRUE)
+		if(BURN)
+			playsound(src, 'sound/items/welder.ogg', 80, TRUE)
+
+/obj/structure/moth/cocoon/Destroy()
+	if(!preparing_to_emerge)
+		visible_message("<span class='danger'>[src] splits open from within!</span>")
+		for(var/mob/living/carbon/human/H in contents)
+			log_game("[key_name(H)] has emerged from their cocoon with the nutrition level of [H.nutrition][H.nutrition <= NUTRITION_LEVEL_STARVING ? ", now starving" : ""]")
+	else
+		visible_message("<span class='danger'>[src] is torn open, harming the Mothperson within!</span>")
+		for(var/mob/living/carbon/human/H in contents)
+			H.drowsyness += COCOON_HARM_AMOUNT / 2
```

Comment:
is there a proc for these? if not ew but fine

also check for the status effect here